src/util/log/fd_log.c

```diff
@@ -1234,6 +1234,138 @@ fd_log_private_halt( void ) {
 //FD_LOG_INFO(( "fd_log: halt success" )); /* Log not online anymore */
 }
 
+#include <sys/resource.h>
+
+ulong
+fd_log_private_main_stack_sz( void ) {
+
+  /* We are extra paranoid about what rlimit returns and we don't trust
+     environments that claim an unlimited stack size (because it just
+     isn't unlimited ... even if rlimit says otherwise ... which it will
+     if a user tries to be clever with a "ulimit -s unlimited" ... e.g.
+     tile0's stack highest address is at 128 TiB-4KiB typically on
+     modern Linux and grows down while the text / data / heap grow up
+     from 0B ... so stack size is practically always going to be << 128
+     TiB irrespective of any getrlimit claim).  TODO: It looks like
+     pthead_attr_getstack might be getrlimit based under the hood, so
+     maybe just use pthread_attr_getstack here too? */
+
+  struct rlimit rlim[1];
+  int err = getrlimit( RLIMIT_STACK, rlim );
+  if( FD_UNLIKELY( err ) ) {
+    FD_LOG_WARNING(( "fd_log: getrlimit failed (%i-%s)", errno, strerror( errno ) ));
+    return 0UL;
+  }
+
+  ulong stack_sz = (ulong)rlim->rlim_cur;
+  if( FD_UNLIKELY( (rlim->rlim_cur>rlim->rlim_max) | (rlim->rlim_max>RLIM_INFINITY    ) |
+                   (rlim->rlim_cur==RLIM_INFINITY) | (rlim->rlim_cur!=(rlim_t)stack_sz) ) ) {
+    FD_LOG_WARNING(( "fd_log: unexpected stack limits (rlim_cur %lu, rlim_max %lu)",
+                     (ulong)rlim->rlim_cur, (ulong)rlim->rlim_max ));
+    return 0UL;
+  }
+
+  return stack_sz;
+}
+
+/* When pthread_setstack is not used to explicitly set the memory region
+   for a new thread's stack, pthread_create will create a memory region
+   (using either the requested size or a default size).  And, while
+   pthread allows us to set and get the size of the stack region it
+   creates and we can get a pointer into a thread's stack by just
+   declaring a stack variable in that thread and we obviously know where
+   a thread's stack is when we explicitly specify it to pthread create,
+   pthreads does not seem to provide a simple way to get the extents of
+   the stacks it creates.
+
+   But the relationship between a pointer in the stack and the stack
+   extents is non-trival because pthreads will use some of the stack for
+   things like thread local storage (and it will not tell us how much
+   stack was used by that and this is practically only known after
+   linking is complete and then it is not simply exposed to the
+   application).
+
+   Similar applies to the first thread's stack.  We can learn how large
```

Comment:
```suggestion
   Similar uncertainty applies to the first thread's stack.  We can learn how large
```
![image](https://github.com/firedancer-io/firedancer/assets/520236/e3b9f4a2-f124-4478-b699-5319a56a0e30)